### ISSUE TITLE
Value d1windrgr can't be converted to <class 'float'>

### ISSUE THREAD

**Issue by hmmbob** [golles/ha-knmi#58]
### Did you read the instructions?

- [X] I have read the [README](../blob/main/README.md), including the [known-limitations](../blob/main/README.md#known-limitations) section.

### The problem

Since 01:26 this night, my log is overflown with the error in the subject. It's continiously happening until now.

### What version of this integration has the issue?

1.3.4

### What version of Home Assistant Core has the issue?

core-2022.11.4

### Home Assistant log

```shell
2022-11-28 01:06:12.348 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:06:12.357 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:06:42.350 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:06:42.352 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:07:12.352 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:07:12.353 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:07:42.351 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:07:42.352 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:08:12.352 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:08:12.353 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:08:42.354 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:08:42.356 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:09:12.355 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:09:12.356 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:09:42.355 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:09:42.357 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:10:12.357 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:10:12.359 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:10:42.358 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:10:42.359 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:11:12.359 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:11:12.360 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:11:42.360 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:11:42.362 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:12:12.361 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:12:12.363 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:12:42.363 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:12:42.365 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:13:12.364 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:13:12.365 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:13:42.365 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:13:42.367 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:14:12.366 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:14:12.368 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:14:42.368 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:14:42.369 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:15:12.369 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:15:12.370 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:15:42.369 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:15:42.371 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:16:12.370 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:16:12.372 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:16:42.371 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:16:42.372 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:17:12.371 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:17:12.372 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:17:42.372 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:17:42.374 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:18:12.373 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:18:12.375 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:18:42.375 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:18:42.376 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:19:12.375 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:19:12.377 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:19:42.376 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:19:42.378 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:20:12.376 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:20:12.378 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:20:42.377 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:20:42.379 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:21:12.377 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:21:12.379 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:21:42.379 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:21:42.381 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:22:12.381 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:22:12.382 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:22:42.381 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:22:42.383 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:23:12.382 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:23:12.383 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:23:42.383 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:23:42.385 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:24:12.384 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:24:12.385 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:24:42.385 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:24:42.387 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:25:12.387 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:25:12.388 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:25:42.387 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:25:42.389 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:26:12.389 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:26:12.391 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:26:42.391 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:26:42.392 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:27:12.391 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:27:12.393 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:27:42.392 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:27:42.394 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:28:12.393 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:28:12.395 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:28:42.395 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:28:42.396 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:29:12.396 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:29:12.397 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:29:42.396 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:29:42.397 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:30:12.398 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:30:12.399 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:30:42.398 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:30:42.400 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:31:12.400 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:31:12.401 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:31:42.401 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:31:42.402 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:32:12.402 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
2022-11-28 01:32:12.404 WARNING (MainThread) [custom_components.knmi] Value d1windrgr can't be converted to <class 'float'>
```

### Additional information

I'm not even using this value actively.

Current contents of `weather.knmi_home`:

```yaml
temperature: 7.5
temperature_unit: °C
humidity: 97
pressure: 1009.6
pressure_unit: hPa
wind_bearing: 180
wind_speed: 10.8
wind_speed_unit: km/h
visibility: 5
visibility_unit: km
precipitation_unit: mm
forecast:
  - datetime: '2022-11-27T23:00:00+00:00'
    condition: cloudy
    templow: 6
    temperature: 10
    precipitation_probability: 0
    wind_bearing: 225
    wind_speed: 11
    wind_speed_bft: 2
    sun_chance: 0
  - datetime: '2022-11-28T23:00:00+00:00'
    condition: cloudy
    templow: 4
    temperature: 8
    precipitation_probability: 30
    wind_bearing: null
    wind_speed: 7
    wind_speed_bft: 2
    sun_chance: 20
  - datetime: '2022-11-29T23:00:00+00:00'
    condition: cloudy
    templow: 5
    temperature: 7
    precipitation_probability: 30
    wind_bearing: 44
    wind_speed: 7
    wind_speed_bft: 2
    sun_chance: 10
attribution: KNMI Weergegevens via https://weerlive.nl
```

**Comment by golles** [comment]
Hi @hmmbob,

No worries about this, this is just a warning, not an error, to log something is wrong with the data from the API. The field `d1windrgr` is "missing" (empty string). This logging feature is introduced recently in #54.

Snippet from the API response:
<img width="143" alt="Screenshot 2022-11-28 at 09 51 52" src="https://user-images.githubusercontent.com/76483909/204235031-2d2895ca-9e2b-474f-9eff-206c361a6672.png">

`d1windrgr` is being used in `weather.knmi_home.forecast[1]. wind_bearing` (pseudo code)

**Comment by golles** [comment]
I think `warning` is correct in such a situation, due to the missing data there is something wrong with the state and it's good to inform the user Scott this with a warning. 

I think I do understand why the field is empty, as the field `d1windr` = `VAR` which means the wind dir is variable.
I'll fix this asap.

**Comment by hmmbob** [comment]
Thanks, laat maar weten als ik even moet testen

**Comment by golles** [comment]
> Thanks, laat maar weten als ik even moet testen

There is a fix in this branch that you can test: `58-value-d1windrgr-cant-be-converted-to-class-float` if you want, you can test that. I've got it running for half a day and will probably release this in the coming 1-2 days when everything looks fine.

**Comment by golles** [comment]
I've had this fix running for a few days. but haven't seen any VAR wind speeds during those days. I think the fix is sufficient, there was a test added to cover the specific case.
It has been released in 1.3.5, feel free to reopen the issue when it happens again.
